### ISSUE TITLE
FhirDateTime toDateTimeOffset exception when system locale does not use colon as time separator

### ISSUE THREAD

**Issue by plindman** [ewoutkramer/fhir-net-api#256]
FhirDateTime toDateTimeOffset throws error when string contains other character than colon as time separator. FhirDateTime.Now on the other hand creates a locale-specific string (time part for instance like "T19.10.07"). Should use culture invariant versions of DateTime methods to ensure consistently correct format.

```
DateTimeOffset.Now.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", System.Globalization.CultureInfo.InvariantCulture));
DateTimeOffset.Parse("2016-10-10T19:00:02+01:00", System.Globalization.CultureInfo.InvariantCulture));

```

**Comment by brianpos** [comment]
Thanks for reporting it, will get it resolved.

**Comment by ewoutkramer** [comment]
I have always assumed that if I gave it a format, it would use that format, but as usual with datetime types, this is more complex than expected ;-)

**Comment by ewoutkramer** [comment]
I think I fixed this in 37511bb6c56ae8c37af70bf3b8bafceea5351658, but I wouldn't mind if one of you checks what I did there.

**Comment by ewoutkramer** [comment]
Oops, i meant this one: 365fc99a0c48db68e7f78fa13e2f3243d8791091

**Comment by plindman** [comment]
Seems OK. The key is to use the same library both ways when converting. 

Mind that [XmlConvert.ToString output format is undocumented](https://msdn.microsoft.com/en-us/library/bb517387(v=vs.110).aspx), so you may want to [explicitly set the format](https://msdn.microsoft.com/en-us/library/bb552356(v=vs.110).aspx) ; i.e. `XmlConvert.ToString(dto, FMT_FULL)`. And use [this when converting back](https://msdn.microsoft.com/en-us/library/bb517525(v=vs.110).aspx); i.e. `XmlConvert.ToDateTimeOffset(value, FMT_FULL);`

Alternatively to XmlConvert you can do this:
``` string dt_inv = dt.ToString(FMT_FULL, CultureInfo.InvariantCulture);
DateTimeOffset dto = DateTimeOffset.ParseExact(dt_inv, FMT_FULL, CultureInfo.InvariantCulture); ```

Here a short .NET program to show both options in action: [GIST](https://gist.github.com/plindman/93b776131ae78ec0e2736314bbd593c8)

Locally I get the following output: 
Datetime conversion ...
DateTimeOffset ToString:
No culture:2016-10-17T18.35.46+03:00
Invariant:2016-10-17T18:35:46+03:00
Finnish with dot:2016-10-17T18.35.46+03:00
British with colon:2016-10-17T18:35:46+03:00
XmlConvert:
No format:2016-10-17T18:35:46.4745942+03:00
Full format:2016-10-17T18:35:46+03:00
DateTimeOffset Parse ...
XmlConvert Parse ...
Parse all OK

**Comment by ewoutkramer** [comment]
Ok, double check and fix if necessary.

**Comment by ewoutkramer** [comment]
Wow, there was quite some inconsistency here, sometimes I used XmlConvert, other times PrimitiveTimeConverter.  I have tried to locate all the uses of XmlConvert and ToString() and send all invocations to the central PrimitiveTypeConverter class, which does it correctly.

I could *not* use XmlConverter with the FMT_FULL parameter (nor the other suggestions), since I depend on XmlConverters conversion of /partial/ datetimes (i.e. it will convert to midnight if no time is supplied). 

So, although MS does not explicitly document XmlConvert to be non-locale dependent, we must assume it is, since using anything else than ":" in a time is not compliant with xsd:dateTime.

Fixed in 15c9b8b4f7fbe44925c026415092ca6809d86bf3
